ticker_service.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"sync"
+	"time"
+)
+
+type TickerService struct {
+	mu           sync.Mutex
+	ds           *DeviceService
+	measurements chan<- Measurement
+	stop         chan bool
+	isRunning    bool
+}
+
+func (ts *TickerService) Start() error {
+	ts.mu.Lock()
+	defer ts.mu.Unlock()
+
+	if ts.isRunning {
+		log.Println("tickers started already")
+		return nil
+	}
+
+	devices, err := ts.ds.GetAll(0, 0)
+	if err != nil {
+		return errors.New("failed to start measurements sending")
```

Comment:
you are swallowing the original error here. It is good practice to override the error before retruning it to the user but you are loosing here the information about the real issue. Please just log it.